main/src/main/java/com/kenshoo/pl/simulation/DualRunSimulator.java

```
@@ -0,0 +1,139 @@
+ *
+ * @param <E> EntityType
+ * @param <ID> ID type
+ */
+public class DualRunSimulator<E extends EntityType<E>, ID extends Identifier<E>> {
+
+    private final Collection<EntityField<E, ?>> inspectedFields;
+    private final ChangeFlowConfig.Builder<E> flowToSimulate;
+    private final PersistenceLayer<E> pl;
+    private final ActualResultFetcher<E, ID> actualResultFetcher;
+    private final ResultComparator<E, ID> resultComparator;
+
+    public DualRunSimulator(PLContext plContext, ChangeFlowConfig.Builder<E> flowToSimulate, Collection<EntityField<E, ?>> inspectedFields) {
+        this.inspectedFields = inspectedFields;
+        this.flowToSimulate = flowToSimulate;
+        this.actualResultFetcher = new ActualResultFetcher<>(new EntitiesFetcher(plContext.dslContext()), inspectedFields);
+        this.pl = new PersistenceLayer<>(plContext);
+        this.resultComparator = new ResultComparator<>(inspectedFields);
+    }
+
+    /**
+     * Note that auto-increment is not supported for simulation. IDs must be provided in order to compare simulated
+     * commands with actual mutations.
+     *
+     * @param databaseMutator
+     * @param commandsToSimulate
+     * @return
+     */
+    public List<ComparisonMismatch<E, ID>> runCreation(
+            ActualDatabaseMutator<E, ID> databaseMutator,
+            Collection<? extends Pair<ID, ? extends CreateEntityCommand<E>>> commandsToSimulate) {
+
+        final var plFlow = flowToSimulate.withoutOutputGenerators().build();
+
+        populateIdsInCreateCommands(commandsToSimulate);
+
+        final var simulatedResults = seq(pl.create(values(commandsToSimulate), plFlow).getChangeResults())
+                .zip(commandsToSimulate)
+                .map(r -> new SimulatedResult<>(r.v1.getCommand(), r.v2.getKey(), r.v1.getErrors()))
+                .collect(toList());
+
+        final var actualErrors = seq(databaseMutator.run()).toMap(__ -> __.getId());
+
```

@galkoren 
Catch all exceptions, write warnings to log, and return empty result.